### DISCUSSION TITLE
Loading full objects when reading open API document

### DISCUSSION THREAD

**Discussion by salmanbabri** [microsoft/OpenAPI.NET#1244] [Q&A] [answered]
Hi, I've tried the example given in the brief documentation.

```
var httpClient = new HttpClient
{
    BaseAddress = new Uri("https://raw.githubusercontent.com/OAI/OpenAPI-Specification/")
};

var stream = await httpClient.GetStreamAsync("master/examples/v3.0/petstore.yaml");
var openApiDocument = new OpenApiStreamReader().Read(stream, out var diagnostic);
```

When I inspect the returned object, it is mostly empty, with no tags, empty OpenApiItem object etc.  Is there a way to load a fully detailed object with all the information present in the original yaml/json?

**Comment by baywet** [comment, marked as answer]
Hi @salmanbabri

Thank you for using the SDK and for reaching out.

That's most likely because the full URL is actually `https://raw.githubusercontent.com/OAI/OpenAPI-Specification/refs/heads/main/_archive_/schemas/v3.0/pass/petstore.yaml`

Ignoring the fact that `master/examples/v3.0/petstore.yaml` changed to `main/_archive_/schemas/v3.0/pass/petstore.yaml`, due to branch rename and the repo moving things around, your example is missing the `refs/heads` section.

To answer your initial question, no the library doesn't let you access the original stream, or the JsonNode tree it looked at. Even with versions 2.X or 3.X. However the diagnostics should give you an idea of what's going on. Additionally in your example you have access to the HTTP response (headers, body, etc...).

Let us know if you have any additional comments or questions.
